hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/Route.java

```diff
@@ -81,15 +85,27 @@ public void configure() {
                 .handled(true)
                 .id("ValidationException");
         
+        onCompletion().choice()
+        .when(header("isFileDropsAllowed")
+        		.isEqualTo("Y"))
+        .bean(V2FileDrops.class).id("V2FileDrops");
```

Comment:
This only seems to work when the route actually completes successfully (not in the case of an exception). I think we want multiple specific calls to the file drops bean each time we need a file drop.

So we'd have one right at the start, one at the end, and one on any exception handlers. We could use WireTap so that it doesn't add any additional processing time to the transaction: https://camel.apache.org/components/3.4.x/eips/wireTap-eip.html 

I think that would be a little more clear to future developers too rather than storing the request message as an exchange property.

Reply:
Incorporated changes. Please review.